pkg/controller/rds/dbclusterparametergroup/setup.go

```diff
@@ -202,3 +214,29 @@ func (c *custom) getDBEngineVersion(ctx context.Context, selector *svcapitypes.D
 	}
 	return resp.DBEngineVersions[0], nil
 }
+
+func (c *custom) parametersToUpdate(cr *svcapitypes.DBClusterParameterGroup, current []*svcsdk.Parameter) []svcapitypes.Parameter {
+	var parameters []svcapitypes.Parameter
+	observed := make(map[string]svcsdk.Parameter, len(current))
+
+	for _, p := range current {
+		observed[awsclients.StringValue(p.ParameterName)] = *p
+	}
+
+	// compare CR with currently set Parameters
+	for _, v := range cr.Spec.ForProvider.Parameters {
+		existing, ok := observed[awsclients.StringValue(v.ParameterName)]
+
+		if !ok {
+			parameters = append(parameters, v)
+			continue
+		}
+
+		if awsclients.StringValue(existing.ParameterValue) != awsclients.StringValue(v.ParameterValue) ||
+			awsclients.StringValue(existing.ApplyMethod) != awsclients.StringValue(v.ApplyMethod) {
```

Comment:
I think there are more fields to consider than just the value and the apply method: https://github.com/crossplane-contrib/provider-aws/blob/f0163fa10ab2856dcc9429e0c123d70b0c863e20/apis/rds/v1alpha1/zz_types.go#L1263

I believe `cmp.Equal` can help here.

Reply:
Thanks for your review @MisterMX, this is actually just re-purposing the existing logic from `isUpToDate()`, however I can certainly expand it - will take a look

Reply:
@MisterMX  After looking into this a little further I believe the only parameters we can actually set are `ParameterValue` and `ApplyMethod`. The other fields seem to be for describe operations only, when testing locally any change to other fields defined in the struct do nothing to the external resource.

The documentation here seems to confirm this - https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_ModifyDBClusterParameterGroup.html

I believe the reason for this may be that the `parameters` data type is shared between `ModifyDBParameterGroup` and `DescribeDBParameters`  as stated here - https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_Parameter.html